torchaudio/datasets/legacy/vctk.py

```diff
@@ -0,0 +1,243 @@
+
+        for root, _, fnames in sorted(os.walk(d)):
+            for fname in fnames:
+                if is_audio_file(fname):
+                    path = os.path.join(root, fname)
+                    item = path
+                    audios.append(item)
+    return audios
+
+
+def read_audio(fp, downsample=True):
+    if downsample:
+        E = torchaudio.sox_effects.SoxEffectsChain()
+        E.set_input_file(fp)
+        E.append_effect_to_chain("gain", ["-h"])
+        E.append_effect_to_chain("channels", [1])
+        E.append_effect_to_chain("rate", [16000])
+        E.append_effect_to_chain("gain", ["-rh"])
+        E.append_effect_to_chain("dither", ["-s"])
+        sig, sr = E.sox_build_flow_effects()
+    else:
+        sig, sr = torchaudio.load(fp)
+    sig = sig.contiguous()
+    return sig, sr
+
+
+def load_txts(dir):
+    """Create a dictionary with all the text of the audio transcriptions."""
+    utterences = dict()
+    dir = os.path.expanduser(dir)
+    for target in sorted(os.listdir(dir)):
+        d = os.path.join(dir, target)
+        if not os.path.isdir(d):
+            continue
+
+        for root, _, fnames in sorted(os.walk(d)):
+            for fname in fnames:
+                if fname.endswith(".txt"):
+                    with open(os.path.join(root, fname), "r") as f:
+                        fname_no_ext = os.path.basename(
+                            fname).rsplit(".", 1)[0]
+                        utterences[fname_no_ext] = f.readline()
+    return utterences
+
+
+class VCTK(data.Dataset):
+    r"""`VCTK <http://homepages.inf.ed.ac.uk/jyamagis/page3/page58/page58.html>`_ Dataset.
+    `alternate url <http://datashare.is.ed.ac.uk/handle/10283/2651>`_
+
+    Args:
+        root (str): Root directory of dataset where ``processed/training.pt``
+            and  ``processed/test.pt`` exist.
+        downsample (bool, optional): Whether to downsample the signal (Default: ``True``)
+        transform (Callable, optional): A function/transform that takes in an raw audio
+            and returns a transformed version. E.g, ``transforms.Spectrogram``. (Default: ``None``)
+        target_transform (callable, optional): A function/transform that takes in the
+            target and transforms it. (Default: ``None``)
+        download (bool, optional): If true, downloads the dataset from the internet and
+            puts it in root directory. If dataset is already downloaded, it is not
+            downloaded again. (Default: ``True``)
+        dev_mode(bool, optional): If true, clean up is not performed on downloaded
+            files.  Useful to keep raw audio and transcriptions. (Default: ``False``)
+    """
+    raw_folder = 'vctk/raw'
+    processed_folder = 'vctk/processed'
+    url = 'http://homepages.inf.ed.ac.uk/jyamagis/release/VCTK-Corpus.tar.gz'
+    dset_path = 'VCTK-Corpus'
+
+    def __init__(self, root, downsample=True, transform=None, target_transform=None, download=False, dev_mode=False):
```

Comment:
I assume the transforms key are still part of the current API, otherwise, in the future, I'd suggest we not make transforms necessary for datasets at all, but instead provide an easy block that combines transforms.